test/extended/util/client.go

```diff
@@ -558,6 +558,13 @@ func (c *CLI) AdminDynamicClient() dynamic.Interface {
 }
 
 func (c *CLI) NewPrometheusClient(ctx context.Context) prometheusv1.API {
+	microShift, err := IsMicroShiftDeployment(c)
```

Comment:
This is not the right place for checking is-microshift condition. Every test invoking `NewPrometheusClient` needs to decide whether it can fallback to a different mechanism in case the Prometheus is not installed. Or whether the right action is to skip a test.

Reply:
Changed this to a more adequate place.